src/java/org/apache/cassandra/repair/messages/RepairMessage.java

```diff
@@ -165,13 +188,37 @@ public void onFailure(InetAddressAndPort from, RequestFailureReason failureReaso
                                                          backoff.computeWaitTime(attempt), backoff.unit());
                             return;
                         }
+                        maybeRecordRetry(failureReason);
                         finalCallback.onFailure(from, failureReason);
                         return;
                     default:
                         throw new AssertionError("Unknown error handler: " + allowed);
                 }
             }
 
+            private void maybeRecordRetry(@Nullable RequestFailureReason reason)
+            {
+                if (attempt <= 0)
+                    return;
+                // we don't know what the prefix kind is... so use NONE... this impacts logPrefix as it will cause us to use "repair" rather than "preview repair" which may not be correct... but close enough...
+                String prefix = PreviewKind.NONE.logPrefix(request.parentRepairSession());
+                RepairMetrics.retry(verb, attempt);
+                if (reason == null)
+                {
+                    noSpam.info("{} Retry of repair verb " + verb + " was successful after {} attempts", prefix, attempt);
+                }
+                else if (reason == RequestFailureReason.TIMEOUT)
+                {
+                    noSpam.warn("{} Timeout for repair verb " + verb + "; could not complete within {} attempts", prefix, attempt);
+                    RepairMetrics.retryTimeout(verb);
+                }
+                else
+                {
+                    noSpam.warn("{} Failure for repair verb " + verb + "; could not complete within {} attempts", prefix, attempt);
```

Comment:
Is it worth logging the `reason` as well?

Reply:
switched to `WARN  [main] 2023-10-26 14:48:08,084 NoSpamLogger.java:107 - [repair #58395110-7449-11ee-b645-8b78869f39eb] UNKNOWN failure for repair verb VALIDATION_REQ; could not complete within 1 attempts`